platform/ui/src/components/HeaderPatientInfo/HeaderPatientInfo.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '@ohif/ui';
+import { utils } from '@ohif/core';
+import { useAppConfig } from '@state';
+
+const { formatDate, formatPN } = utils;
+
+function usePatientInfo(servicesManager) {
+  const [patientInfo, setPatientInfo] = useState({
+    PatientName: '',
+    PatientID: '',
+    PatientSex: '',
+    PatientAge: '',
+    PatientDOB: '',
+  });
+  const [isMixedPatients, setIsMixedPatients] = useState(false);
+
+  const { displaySetService, viewportGridService, cornerstoneViewportService } =
+    servicesManager.services;
+
+  const { activeViewportId, viewports } = viewportGridService.getState();
+
+  const checkMixedPatients = activeViewportPatientID => {
+    const { viewports } = viewportGridService.getState();
+    let isMixedPatients = false;
+    viewports.forEach(viewport => {
+      const displaySetInstanceUID = viewport.displaySetInstanceUIDs[0];
+      if (!displaySetInstanceUID) {
+        return;
+      }
+      const displaySet = displaySetService.getDisplaySetByUID(displaySetInstanceUID);
+      const instance = displaySet?.instances?.[0] || displaySet?.instance;
+      if (!instance) {
+        return;
+      }
+
+      if (instance.PatientID !== activeViewportPatientID) {
+        isMixedPatients = true;
+      }
+    });
+    setIsMixedPatients(isMixedPatients);
+  };
+
+  const updatePatientInfo = (viewportId = null) => {
+    const state = viewportGridService.getState();
+    const activeViewportId = viewportId || state.activeViewportId;
+    const activeViewport = state.viewports.get(activeViewportId);
+    const displaySetInstanceUID = activeViewport?.displaySetInstanceUIDs[0];
+
+    if (!displaySetInstanceUID) {
+      return;
+    }
+    const displaySet = displaySetService.getDisplaySetByUID(displaySetInstanceUID);
+    const instance = displaySet?.instances?.[0] || displaySet?.instance;
+
+    if (!instance) {
+      return;
+    }
+
+    setPatientInfo({
+      PatientID: instance.PatientID || '',
+      PatientName: instance.PatientName ? formatPN(instance.PatientName.Alphabetic) : '',
+      PatientSex: instance.PatientSex || '',
+      PatientAge: instance.PatientAge || '',
+      PatientDOB: formatDate(instance.PatientBirthDate) || '',
+    });
+
+    checkMixedPatients(instance.PatientID || '');
+  };
+
+  useEffect(() => {
+    const subscriptions = [
+      viewportGridService.subscribe(
+        viewportGridService.EVENTS.ACTIVE_VIEWPORT_ID_CHANGED,
+        event => updatePatientInfo(event.viewportId) // Update the toolbar when the active viewport changes
+      ).unsubscribe,
+      cornerstoneViewportService.subscribe(
+        cornerstoneViewportService.EVENTS.VIEWPORT_DATA_CHANGED,
+        () => updatePatientInfo() // Update the toolbar when the viewport data changes
+      ).unsubscribe,
+    ];
+
```

Comment:
this can be re-written as 

const events = [... ,...]
events.forEach(event, () => {})